02_knight_and_princess/roman_kuprii/main.cpp

```diff
@@ -112,27 +103,49 @@ class Graph {
     return res;
   }
 
-  void add_vertex() { vertices_.emplace_back(vertices_.size()); }
+  void add_vertex() { vertices_.emplace_back(get_next_vertex_id()); }
+
+  bool is_connected(const VertexId& from_vertex_id,
+                    const VertexId& to_vertex_id) const {
+    vector<EdgeId> from_vertex_edges_ids =
+        vertices_[from_vertex_id].get_edges_ids();
+    vector<EdgeId> to_vertex_edges_ids =
+        vertices_[to_vertex_id].get_edges_ids();
```

Comment:
1. `const`?
1. Чем копировать данные, лучше использовать ссылку. 
1. Можно чуть упростить запись используя `auto`
```suggestion
    const auto& from_vertex_edges_ids =
        vertices_[from_vertex_id].get_edges_ids();
    const auto& to_vertex_edges_ids =
        vertices_[to_vertex_id].get_edges_ids();
```